src/transformers/models/llama/modeling_llama.py

```diff
@@ -754,6 +818,10 @@ def _init_weights(self, module):
             Indices depicting the position of the input sequence tokens in the sequence. Contrarily to `position_ids`,
             this tensor is not affected by padding. It is used to update the cache in the correct position and to infer
             the complete sequence length.
+        position_embeddings (`Tuple[torch.FloatTensor, torch.FloatTensor]`, *optional*):
+            Tuple containing the cosine and sine positional embeddings of shape `(batch_size, seq_len, head_dim)`,
+            with `head_dim` being the embedding dimension of each attention head. This input is used to dynamically
+            overwrite the default positional embeddings.
```

Comment:
For my own understanding, under what cases are we expecting this to be passed in? i.e. when would someone want or need to override the position embeddings when calling the model?

Reply:
The rope preparation functions assume RoPE is fully parameterized by `inv_frequency` and `attention_scaling`. Our past experience tells us users will find new ways to shake things up.

This should make tinkering much easier :)

Reply:
hmmmm.... it also adds a another dimension to how models can be used --> increasing space of possible argument interactions and code behaviours. If it's not needed or requested I'd rather we didn't add this atm

Reply:
removed ✅